beacon_node/http_api/src/block_id.rs

```diff
@@ -47,7 +48,26 @@ impl BlockId {
                         ))
                     })
                 }),
-            CoreBlockId::Root(root) => Ok(*root),
+            CoreBlockId::Root(root) => {
+                if root == &Hash256::zero() {
```

Comment:
```suggestion
                // This check ensures that we match a precedent set by Teku.
                if root == &Hash256::zero() {
```

Just to help prevent someone from removing it in the future.